pkgs/fc/qemu/clean-logs.py

```diff
@@ -0,0 +1,54 @@
+"""
+A helper to clean up log files from fc.qemu.
+
+The logrotate targets aren't smart enough because we have VMs that move around
+and thus would end up with missing files, but `missingok` doesn't help if the
+pattern doesn't match any more.
+
+In addition we may have files that have not been written to for a long time but
+do correspond to a VM still running here.
+
+So this script deletes log files, iff:
+
+1. the file is not open, AND
+2. it is older than 14 days.
+
+"""
+import datetime
+import pathlib
+import re
+import subprocess
+import time
+
+MAX_AGE = datetime.timedelta(days=14)
+VM_LOG_DIR = pathlib.Path("/var/log/vm")
+VM_LOG_NAME_PATTERN = re.compile(r"[0-9a-zA-Z]+\.log")
+
+# Find logs of running VMs
+
+print("Checking open files ...")
+
+lsof = subprocess.run(["lsof", "+D", VM_LOG_DIR], capture_output=True)
```

Comment:
We already discussed out-of-band that this command is wrong and its output is not suited to the following parsing code.